dicts/dicts.gobra

```diff
@@ -0,0 +1,605 @@
+/*
+  This file is part of gobra-libs which is released under the MIT license.
+  See LICENSE or go to https://github.com/viperproject/gobra-libs/blob/main/LICENSE
+  for full license details.
+
+  This file is inspired by the standard libraries and axiomatisations of the following verifiers:
+  - dafny-lang/dafny: https://github.com/dafny-lang/dafny/blob/master/Source/DafnyStandardLibraries/src/Std/Collections/Map.dfy
+  - verus-lang/verus:
+    - https://github.com/verus-lang/verus/blob/main/source/pervasive/map_lib.rs
+    - https://github.com/verus-lang/verus/blob/main/source/pervasive/map.rs
+  - viperproject/silicon: https://github.com/viperproject/silicon/blob/master/src/main/resources/dafny_axioms/maps.vpr
+  - why3:
+    - https://www.why3.org/stdlib/fmap.html
+    - https://www.why3.org/stdlib/map.html
+*/
+
+// TODO (at the end) Organize definitions: Remove duplicates, merge similar ones, place related ones together
+
+// QUES I plan on making the naming of variables consistent (e.g. keys and values: x/y, or k/v, or key/value);
+// Is there any convention that is particularly "good"?
+
+// This package defines lemmas for mathematical maps (aka dictionaries) commonly used
+// in specifications.
+package dicts
+
+import "util"
+import "sets"
+
+/// Dafny
+// Skip Get: No Option<>
```

Comment:
These are the notes I usually delete right before the PR is pushed. It just means that I skipped the [Get function from Dafny](https://github.com/dafny-lang/dafny/blob/ce761e745c641d41fdba30b644f9df9eec476ff1/Source/DafnyStandardLibraries/src/Std/Collections/Map.dfy#L17), because I thought there was no option type.